core/src/main/java/org/apache/james/core/quota/QuotaComponent.java

```diff
@@ -0,0 +1,45 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import org.apache.james.core.StringWrapper;
+
+public class QuotaComponent extends StringWrapper {
```

Comment:
Isn't
```
public class QuotaComponent {
    private final String value;
    // constructor, getter, equals, hashcode, toString
}
```
Enough?

rf: https://github.com/apache/james-project/pull/1666#discussion_r1282614846

Reply:
No it is not enough. We should restrict the value of QuotaComponent within three values mailbox, sieve, jmapUploads

Reply:
The way how we blueprint the `QuotaComponent` class is not related to `restrict the value of QuotaComponent`, no?

How is that related to `StringWrapper` usage BTW?

Reply:
+1 with Quan. The `StringWrapper` is a workaround for a language limitation.

Complex hierarchy should be avoided and 'flat' POJOs shall be prefered.

If you really want to limit the boiler plate here then you could consider adding in some scala. Or waiting a Java 17+ switch.

Reply:
deleted